### PR TITLE
Make the correlation header name configurable

### PR DESCRIPTION
**Pull request by asolove**
Please let me know if you see anything surprising or incorrect here. I'm happy to change the way the configuration happens or any other details I may have done less than optimally.

### CODE REVIEW

**Comment by asolove** [comment]
Looks like the Travis build is failing by not being able to satisfy some of the `require`s, which are not included in the gemspec. Removing them, the tests pass and don't seem to be missing out on anything in particular.

**Comment by asolove** [comment]
Ah ok now that's just my failure to think about old rubies. Will fix tomorrow.

File: lib/marlowe/middleware.rb [hunk 1]
```diff
@@ -4,23 +4,33 @@
 
 module Marlowe
   # Marlowe correlation id middleware. Including this into your
-  # middleware stack will add a 'Correlation-Id' header as an incoming
+  # middleware stack will add a correlation id header as an incoming
   # request, and save that id in a request session variable.
 
+  # Name of the default header to look for and put the correlation id in.
+  CORRELATION_HEADER = 'Correlation-Id'
```
Comment (halostatue) [inline lib/marlowe/middleware.rb]: Please `.freeze` this string.

**Comment by thefury** [comment]
Thanks very much!

**Comment by asolove** [comment]
Thank you! These small changes were a lot easier than writing it ourselves.

**Pull request merged**